File: $_algotsu-kaizen/problemo-set/minimum-bit-flips-to-convert-number.c

int minBitFlips(int start, int goal) {
    unsigned int bit = 0b1;
    int kai_bits = 0b0, max = start > goal ? start : goal;

    while (bit <= max) {
        if ((goal & bit) ^ (start & bit))
            kai_bits++;
        bit = bit << 0b1;
    }

    return kai_bits;
}

File: $_algotsu-kaizen/problemo-set/minimum-bit-flips-to-convert-number.rs
impl Solution {

    pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
        let mut bit: u32 = 0b1;
        let mut kai_bits: u32 = 0b0;
        let max = if (start > goal) { start as u32 } else { goal as u32 };

        while bit <= max {
            if ((goal as u32 & bit) ^ (start as u32 & bit)) != 0b0 {
                kai_bits += 0b1;
            }

            bit = bit << 0b1;
        }

        return kai_bits as i32;
    }
}
